htlcswitch/payment_result.go

```diff
@@ -26,6 +27,18 @@ var (
 	// ErrPaymentIDAlreadyExists is returned if we try to write a pending
 	// payment whose paymentID already exists.
 	ErrPaymentIDAlreadyExists = errors.New("paymentID already exists")
+
+	// ErrAttemptResultNotAvailable is returned if we try to get a result
+	// for a pending payment whose result is not yet available.
+	ErrAttemptResultNotAvailable = errors.New(
+		"attempt result not yet available",
+	)
+)
+
+const (
+	// pendingHtlcMsgType is a custom message type used to represent a
+	// pending HTLC in the network result store.
+	pendingHtlcMsgType lnwire.MessageType = 32768
```

Comment:
how did you come up with this number ?

Reply:
Just picked the first number in the range for custom messages listed [here](https://github.com/lightningnetwork/lnd/blob/v0.20.0-beta/lnwire/custom.go#L12). Whatever we select, I wonder if we need to somehow prevent it from being used by the SendCustomMessage RPC [endpoint](https://github.com/lightningnetwork/lnd/blob/v0.20.0-beta/server.go)?

Reply:
I don't think we need to guard against this, only msg of a certain type make it to the result store, so I think we are fine.